### DISCUSSION TITLE
LOCAL ID of the Mirth channel

### DISCUSSION THREAD

**Discussion by mpapini64** [nextgenhealthcare/connect#5387] [Q&A] [answered]
Hi everyone, I need to know in the transformer the LOCAL ID of the channels read with the function "ChannelUtil.getChannelIds ();"
The ChannelUtil gives me the name of the channel, its alphanumeric ID, the various states, but I need to know the Local ID of the channel.
A solution could be to open the Mirth db and look for it in the D_CHANNELS, ok but then I would like to know if it is possible to know which db is opened ...
The solution hoped for is that the Local ID is available directly from Mirth.
Can you help me?

**Comment by jonbartels** [comment]
Get a ChannelController reference with `ControllerFactory.getFactory().createChannelController();` 

Then call either ChannelController method:
```    public abstract Channel getChannelById(String channelId);

    public abstract Channel getChannelByName(String channelName);
```
The Channel object exposes `getLocalChannelId()`.

I am curious - what is your use case for having the local channel ID?

**Reply by mpapini64** [reply]
Thanks, then I'll try.
Then I satisfy your curiosity.
I have developed a store procedure that physically deletes the records in error starting from a certain number of days ...
The query is as follows:
**_SELECT DISTINCT TOP 1000 T1.ID FROM D_M (x) T1 INNER JOIN D_MM (x) T2 ON T1.ID = T2.MESSAGE_ID WHERE T1.RECEIVED_DATE <CAST ('+ char (39) + @ dat + char (39) + 'AS DATE) AND T2.ERROR_CODE = 1 ORDER BY T1.ID_**
With the result of this query it creates a table in the DB and puts the MESSAGE_ID and LOCAL_CHANNEL_ID (x) that satisfy the query, then once this table is filled it reads it again and executes many DELETE D_yyyxxx WHERE MESSAGE_ID = zzz
From Mirth I made a Javascript Reader channel that in time launches this storeprocedure for all the configured localids, I wanted to put the channel names in the configurationMap and not the LocalId.
This is because we have several systems all with 3.10.1 / 3.4.2 and the cancellation of error messages, they only implemented it in 4.0.1 (finally, it took so long?)

**Reply by mpapini64** [reply]
I failed with your solution, but I found a way to perform a SELECT using WHERE with a value split by a comma separated string, in this way I pass to the storeprocedure a varchar with the channel names inside comma separated at this point I don't need the Local ID in Mirth anymore!
Thanks again for your interest!

**Comment by JoeFox82** [comment, marked as answer]
You can use this in your filter/transformer (tested with v4.4.1):

`var channelController = com.mirth.connect.donkey.server.controllers.ChannelController.getInstance();`
`var localChannelId = channelController.getLocalChannelId(channelId);`
`logger.info(localChannelId);`

**Reply by pacmano1** [reply]
Or just a one liner.
```JS
logger.info(Packages.com.mirth.connect.donkey.server.controllers.ChannelController.getInstance().getLocalChannelId(channelId))
```
